File: BrownianMotion/sinpar.R
repetir <- 100 # replicas
duracion <- 200 # pasos
datos <-  data.frame() # almacenar resultados 
for (dimension in 1:8) { # variar dimension
    resultado = numeric() # juntar los datos de las replicas
    for (replica in 1:repetir) { # realizar replicas
        pos <- rep(0, dimension) # iniciar en origen
        mayor <- 0 # mayor distancia inicia en cero
        for (t in 1:duracion) { # realizar los pasos
            cambiar <- sample(1:dimension, 1) # elegir eje
            cambio <- 1 
            if (runif(1) < 0.5) {
                cambio <- -1
            }
            pos[cambiar] <- pos[cambiar] + cambio # mover
            mayor <- max(mayor, sqrt(sum(pos**2))) # dist eucl
        }
        resultado = c(resultado, mayor) # almacenar el mayor valor
    }
    datos <- rbind(datos, resultado) # almacenar los resultados de esta dimension
}
png("figuraR.png") # mandar la figura a un archivo
# las filas van a ser los "grupos" de la figura
boxplot(data.matrix(datos), use.cols = FALSE, 
        xlab="Dimensi\u{F3}n", ylab="Distancia m\u{E1}xima", 
        main="Distancia Euclideana")
graphics.off()

File: BrownianMotion/sinpar.py
# -*- coding: utf-8 -*-

from random import random, randint
from math import fabs, sqrt
from random import getrandbits, randint
import matplotlib.pyplot as plt # instalar con pip3

dur = 200
replicas = 100
results = [] # almacen de dimensiones
for dim in range(1, 9): # de uno a ocho
    mayores = [] # almacen de replicas
    for rep in range(replicas):
        pos = [0] * dim # origen
        mayor = 0
        for paso in range(dur):
            eje = randint(0, dim - 1) # elegir eje (van 0, 1, ..., dim - 1)
            if random() < 0.5:
                pos[eje] += 1
            else:
                pos[eje] -= 1
            mayor = max(mayor, sqrt(sum([ p**2 for p in pos ])))
        mayores.append(mayor) # replica terminada
    results.append(mayores) # dimension procesada
fig, ax = plt.subplots()
ax.boxplot(results) 
ax.set_xlabel('Dimensión')
ax.set_ylabel('Distancia máxima')
ax.set_title('Distancia Euclideana')
plt.savefig('figuraPy.png') # mandar a un archivoa
plt.close()
